lib/matplotlib/ticker.py

```diff
@@ -1614,10 +1614,10 @@ def __call__(self):
     def tick_values(self, vmin, vmax):
         if vmax < vmin:
             vmin, vmax = vmax, vmin
-        vmin = self._base.ge(vmin)
         base = self._base.get_base()
-        n = (vmax - vmin + 0.001 * base) // base
-        locs = vmin - base + np.arange(n + 3) * base
+        nmin = round(self._base.ge(vmin) / base)
+        nmax = round(self._base.le(vmax) / base)
+        locs = np.arange(nmin, nmax + 1) * base
```

Comment:
I had a quick look and am pretty sure that the test values are wrong (i.e. they were picked to match the previous algorithm, which returns out of bounds values; they should be changed to match the new algorithm).